cmd/acraserver/listener.go

```diff
@@ -152,12 +277,61 @@ func (server *SServer) handleCommandsConnection(connection net.Conn) {
 }
 
 // start listening commands connections from proxy
-func (server *SServer) StartCommands() {
-	listener, err := network.Listen(server.config.GetAcraAPIConnectionString())
+func (server *SServer) StartCommands(graceful bool) {
+	var listener net.Listener
+	var listenerGraceful *net.TCPListener
+	var err error
+	var connection net.Conn
+	if graceful == true {
+		listenerGraceful, err = network.ListenTCP(server.config.GetAcraAPIConnectionString())
+		if err != nil {
+			log.WithError(err).Errorln("can't start listen command api connections")
+			return
+		}
+		server.socketAPI = listenerGraceful
+	} else {
+		listener, err = network.Listen(server.config.GetAcraAPIConnectionString())
+		if err != nil {
+			log.WithError(err).Errorln("can't start listen command api connections")
+			return
+		}
+	}
+	log.Infof("start listening api %s", server.config.GetAcraAPIConnectionString())
+	for {
+		if graceful == true {
+			connection, err = listenerGraceful.Accept()
+		} else {
+			connection, err = listener.Accept()
+		}
+
+		if err != nil {
+			// log.WithError(err).Errorf("can't accept new connection (%v)", connection.RemoteAddr())
+			log.WithError(err).Errorf("can't accept new connection")
+			continue
+		}
+		// unix socket and value == '@'
+		if len(connection.RemoteAddr().String()) == 1 {
```

Comment:
is it a good idea to check unix socket value like string length?

Reply:
for unix socket RemoteAddr always == "@", LocalAddr has path. For TCP socket and RemoteAddr and LocalAddr has some path like "127.0.0.1:2345" and "127.0.0.1:5431". How you suggest to checK?